R-package/.Rbuildignore

```diff
@@ -2,6 +2,7 @@
 \.gitkeep$
 ^docs$
 ^pkgdown$
+^cran-comments\.md$
```

Comment:
this is exactly what is recommended by http://r-pkgs.had.co.nz/package.html and I see this exact pattern in many other packages, so I don't understand why the Mac and Linux CRAN packages are raising a NOTE about it.

I just added an `rm cran-comments.md`  to  `build-cran-package.sh` to solve that.

Reply:
I do see other files do not include escaping `.` before extension. Maybe it can help?
https://github.com/microsoft/LightGBM/blob/877d58fac74731a5feba8d6ca3bb0cc97d154eb0/R-package/.Rbuildignore#L23-L32

Reply:
I don't know 😕 

`^cran-comments\.md$` is what is recommended in http://r-pkgs.had.co.nz/package.html and I see a mix of escaping and not-escaping in many projects:

* https://github.com/tidyverse/ggplot2/blob/master/.Rbuildignore
* https://github.com/dmlc/xgboost/blob/master/R-package/.Rbuildignore
* https://github.com/jeroen/jsonlite/blob/master/.Rbuildignore

It seems that everyone just fiddles with this file until R CMD check stops complaining 😬